src/ovp_pipeline/layer_schemas.py

```
@@ -0,0 +1,570 @@
+    # Cross-field rules: validators that get the full frontmatter dict.
+    cross_field_rules: tuple[CrossFieldRule, ...] = ()
+
+
+# ---------------------------------------------------------------------------
+# Validators
+# ---------------------------------------------------------------------------
+
+
+def _validate_note_id_matches_stem(value: Any, file_path: Path) -> str | None:
+    if not isinstance(value, str) or not value:
+        return f"note_id missing or empty"
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)